File: SIT-Test_Solution/20-12/C/Baekjoon02225_SumDecomposition.c
// Baekjoon02225_SumDecomposition.c
// https://www.acmicpc.net/problem/2225
#include <stdio.h>

#define MOD 1000000000

int main(void) {
	int N, K;
	scanf("%d %d", &N, &K);

	long long dp[201][201] = { 0LL, };
	for ( int i = 1; i <= K; i++ )
		dp[0][i] = 1;

	for ( int i = 1; i <= N; i++ ) {
		dp[i][1] = 1;
		for ( int j = 2; j <= K; j++ )
			for ( int k = 0; k <= i; k++ )
				dp[i][j] =  (dp[i][j] + dp[k][j - 1]) % MOD;
	}

	printf("%lld\n", dp[N][K]);

	return 0;
}

File: SIT-Test_Solution/20-12/Python3/Baekjoon02225_SumDecomposition.py
# Baekjoon02225_SumDecomposition.py
# https://www.acmicpc.net/problem/2225
MOD = 1_000_000_000

N, K = map( int, input().split() )

dp = [ [ 0 for _ in range( K + 1 ) ] for _ in range( N + 1 ) ]
for i in range( 1, K + 1 ):
	dp[0][i] = 1

for i in range( 1, N + 1 ):
	dp[i][1] = 1
	for j in range( 2, K + 1 ):
		for k in range( i + 1 ):
			dp[i][j] += dp[k][j - 1]
		dp[i][j] %= MOD

print( dp[N][K] )
